packages/esm-active-visits-app/src/active-visits-widget/active-visits.resource.tsx

```diff
@@ -1,23 +1,61 @@
+import dayjs from 'dayjs';
+import useSWR from 'swr';
 import { openmrsFetch, Visit } from '@openmrs/esm-framework';
 
-export interface ActiveVisitRow {
+export interface ActiveVisit {
+  age: string;
   id: string;
-  visitStartTime: string;
-  IDNumber: string;
-  name: string;
+  idNumber: string;
   gender: string;
-  age: string;
-  visitType: string;
+  name: string;
   patientUuid: string;
+  visitStartTime: string;
+  visitType: string;
   visitUuid: string;
 }
 
-export function fetchActiveVisits(abortController: AbortController) {
-  const v =
+export function useActiveVisits() {
+  const customRepresentation =
     'custom:(uuid,patient:(uuid,identifiers:(identifier,uuid),person:(age,display,gender,uuid)),' +
     'visitType:(uuid,name,display),location:(uuid,name,display),startDatetime,' +
     'stopDatetime)';
-  return openmrsFetch<{ results: Visit[] }>(`/ws/rest/v1/visit?includeInactive=false&v=${v}`, {
-    signal: abortController.signal,
+
+  const { data, error, isValidating } = useSWR<{ data: { results: Array<Visit> } }, Error>(
+    `/ws/rest/v1/visit?includeInactive=false&v=${customRepresentation}`,
+    openmrsFetch,
+  );
+
+  const formatDatetime = (startDatetime) => {
+    const todayDate = dayjs();
+    const today =
+      dayjs(startDatetime).get('date') === todayDate.get('date') &&
+      dayjs(startDatetime).get('month') === todayDate.get('month') &&
+      dayjs(startDatetime).get('year') === todayDate.get('year');
+    if (today) {
+      return `Today - ${dayjs(startDatetime).format('HH:mm')}`;
+    } else {
+      return dayjs(startDatetime).format("DD MMM 'YY - HH:mm");
+    }
+  };
+
+  const mapVisitProperties = (visit: Visit): ActiveVisit => ({
+    age: visit?.patient?.person?.age,
+    id: visit.uuid,
+    idNumber: visit?.patient?.identifiers[0]?.identifier,
+    gender: visit?.patient?.person?.gender,
+    name: visit?.patient?.person?.display,
+    patientUuid: visit?.patient?.uuid,
+    visitStartTime: formatDatetime(visit.startDate),
```

Comment:
Ideally, the `formatDatetime` call should happen in the render block. We should try to keep dates in canonical formats until the absolute last moment of code execution, at which point we format them for display. This reduces the chance that someone will attempt e.g. to do math on a display date.